stl/debugger/STL.natvis

```diff
@@ -1955,4 +2000,60 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
     <Expand/>
   </Type>
 
+  <Type Name="std::formatter&lt;*,*&gt;">
+    <DisplayString>{_Specs._Dynamic_width_index}, {_Specs._Dynamic_precision_index}</DisplayString>
+    <Expand>
+      <Item Name="Width index">_Specs._Dynamic_width_index</Item>
+      <Item Name="Precision index">_Specs._Dynamic_precision_index</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::basic_format_parse_context&lt;*&gt;">
+    <DisplayString>{_Format_string}</DisplayString>
+    <Expand>
+      <Item Name="Arguments">_Num_args</Item>
+      <Item Name="Next arg id">_Next_arg_id</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::basic_format_args&lt;*&gt;">
+    <DisplayString>Num args:{_Num_args}</DisplayString>
+    <Expand>
+      <Item Name="Index">_Index_array->_Index</Item>
+      <Item Name="Type">_Index_array->_Type_</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::basic_format_arg&lt;*&gt;">
+    <DisplayString>Active state: {_Active_state}</DisplayString>
+    <Expand>
+      <Item Name="No state">_No_state</Item>
+      <Item Name="Int">_Int_state</Item>
+      <Item Name="UInt state">_UInt_state</Item>
+      <Item Name="Long long state">_Long_long_state</Item>
+      <Item Name="ULong_long_state">_ULong_long_state</Item>
+      <Item Name="Bool state">_Bool_state</Item>
+      <Item Name="Char state">_Char_state</Item>
+      <Item Name="Float state">_Float_state</Item>
+      <Item Name="Double state">_Double_state</Item>
+      <Item Name="Long double state">_Long_double_state</Item>
+      <Item Name="Pointer state">_Pointer_state</Item>
+      <Item Name="CString state">_CString_state</Item>
+      <Item Name="String state">_String_state</Item>
+      <Item Name="Custom state">_Custom_state</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::basic_format_context&lt;*&gt;">
+    <DisplayString>{_Args}</DisplayString>
+    <Expand>
+      <Item Name="Iterator">_OutputIt</Item>
+      <Item Name="Loc">_Loc</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::format_error">
+    <DisplayString>{what()}</DisplayString>

```

Comment:
Have you verified that this works? I think we can't call non-intrinsic functions in natvis.

Reply:
![image](https://user-images.githubusercontent.com/1189138/132760105-12e674b7-adfd-46c5-963a-3462554a40ef.png)
It works but you have to click the blue arrow first

Reply:
Shouldn't we only have a visualizer for `std::exception` and all other exception classes would simply derive their visualizer?

Reply:
updated